tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -3786,35 +3786,47 @@ void TestConvertGather(OpConverterTest* test) {
     int axis;
     std::vector<int> expected_output_dims;
     std::vector<int> expected_output;
+    bool params_is_tensor;
   };
 
   // Input is the same {1, 2, 3, 4, 5, 6} for all cases.
-  const int kGatherOKCases = 7;
+  const int kGatherOKCases = 10;
   const std::vector<CType> params_input = {CType(1), CType(2), CType(3),
                                            CType(4), CType(5), CType(6)};
   TestParams ok_params[kGatherOKCases] = {
       // Vector indices, and output rank is rank(params).
-      TestParams{{1, 2, 3}, {}, {0}, 3, {1, 2, 1}, {1, 4}},
-      TestParams{{1, 2, 3}, {}, {1}, 2, {1, 1, 3}, {4, 5, 6}},
+      TestParams{{1, 2, 3}, {}, {0}, 3, {1, 2, 1}, {1, 4}, true},
+      TestParams{{1, 2, 3}, {}, {1}, 2, {1, 1, 3}, {4, 5, 6}, true},
       // Indices with rank>1, and output rank is rank(params)+rank(indices)-1.
-      TestParams{{1, 2, 3}, {1}, {0}, 3, {1, 2, 1, 1}, {1, 4}},
-      TestParams{{1, 2, 3}, {1}, {1}, 3, {1, 2, 1, 1}, {2, 5}},
-      TestParams{{1, 2, 3}, {1}, {2}, -1, {1, 2, 1, 1}, {3, 6}},
+      TestParams{{1, 2, 3}, {1}, {0}, 3, {1, 2, 1, 1}, {1, 4}, true},
+      TestParams{{1, 2, 3}, {1}, {1}, 3, {1, 2, 1, 1}, {2, 5}, true},
+      TestParams{{1, 2, 3}, {1}, {2}, -1, {1, 2, 1, 1}, {3, 6}, true},
       TestParams{
-          {1, 2, 3}, {3}, {2, 0, 1}, 3, {1, 2, 1, 3}, {3, 1, 2, 6, 4, 5}},
+          {1, 2, 3}, {3}, {2, 0, 1}, 3, {1, 2, 1, 3}, {3, 1, 2, 6, 4, 5}, true},
       TestParams{{3, 2},
                  {2, 2},
                  {0, 0, 1, 0},
                  2,
                  {3, 1, 2, 2},
-                 {1, 1, 2, 1, 3, 3, 4, 3, 5, 5, 6, 5}},
+                 {1, 1, 2, 1, 3, 3, 4, 3, 5, 5, 6, 5},
+                 true},
+      TestParams{{1, 2, 3}, {}, {0}, 0, {1, 2, 3}, {1, 2, 3, 4, 5, 6}, false},
```

Comment:
As mentioned above, I think the reason why these tests didn't fail is the 5th parameter expected_output_dims is not specified correctly. It is the expected **TRT** shape, not the expected **TF** shape, so for example here it should be `{2,3}` not `{1,2,3}`.

Reply:
My understanding looks like that, when I've got params as weights, I should keep the batch dimension in the output of gather operation. 
So in this case 1 is the batch dimension.
Let's make another example:
bs = 2
params dimensions  = (1, 3, 3)  (params is weights)
indices dimensions = ()
axis = 0
In this case, output dimensions should be (2, 3, 3), where 2 is the batch dimension.